dataverk/connectors/kafka_consumer.py

```diff
@@ -0,0 +1,87 @@
+import pandas as pd
+import json
+from kafka import KafkaConsumer
+from collections import Mapping, Sequence
+from enum import Enum
+from datetime import datetime
+
+
+class KafkaFetchMode(Enum):
+    FROM_BEGINNING = "from_beginning"
+    LAST_COMMITED_OFFSET = "last_commited_offset"
+
+
+class DVKafkaConsumer:
+
+    def __init__(self, settings: Mapping, topics: Sequence, fetch_mode: str):
+        """ Dataverk Kafka consumer class
+
+        :param settings: Mapping object containing project settings
+        :param topics: Sequence of topics to subscribe to
+        :param fetch_mode: str describing fetch mode (from_beginning, last_committed_offset)
+        """
+        self._consumer = self._get_kafka_consumer(settings=settings, topics=topics, fetch_mode=fetch_mode)
+        self._read_until_timestamp = self._get_current_timestamp_in_ms()
+
+    def get_pandas_df(self):
+        """ Read kafka topics, commits offset and returns result as pandas dataframe
+
+        :return: pd.Dataframe containing kafka messages read
+        """
+        df = self._read_kafka()
+        self._commit_offsets()
+
+        return df
+
+    def _get_kafka_consumer(self, settings: Mapping, topics: Sequence, fetch_mode: str) -> KafkaConsumer:
+        """ Factory method returning a KafkaConsumer object with desired configuration
+
+        :param settings: Mapping object containing project settings
+        :param topics: Sequence of topics to subscribe
+        :param fetch_mode: str describing fetch mode (from_beginning, last_committed_offset)
+        :return: KafkaConsumer object with desired configuration
+        """
+        if KafkaFetchMode(fetch_mode) is KafkaFetchMode.FROM_BEGINNING:
+            group_id = None
+        elif KafkaFetchMode(fetch_mode) is KafkaFetchMode.LAST_COMMITED_OFFSET:
+            group_id = settings["kafka"].get("group_id", None)
+        else:
+            raise ValueError(f'{fetch_mode} is not a valid KafkaFetchMode. Valid fetch_modes are: '
+                             f'"{KafkaFetchMode.FROM_BEGINNING}" and "{KafkaFetchMode.LAST_COMMITED_OFFSET}"')
+
+        return KafkaConsumer(*topics,
+                             group_id=group_id,
+                             bootstrap_servers=settings["kafka"].get("brokers", "localhost:9092"),
```

Comment:
Hardkodet til å peke på localhost?

Reply:
Hardkodet til å defaulte til localhost dersom ingenting er angitt i settings.json, akkurat som autentisering defaulter til None. Satte det slik for at det skulle være lett å teste kafka lokalt. Usikker på om det er det beste, kanskje burde det kastes exception hvis ikke all nødvendig informasjon er å finne i settings objektet.

Reply:
ok